pkg/election/election.go

```diff
@@ -230,11 +240,23 @@ func (e *Election) campaignLoop(ctx context.Context, session *concurrency.Sessio
 		elec := concurrency.NewElection(session, e.key)
 		ctx2, cancel2 := context.WithCancel(ctx)
 
+		e.campaignMu.Lock()
+		e.cancelCampaign = func() {
+			cancel2()
+			compaignWg.Wait()
+		}
+		e.campaignMu.Unlock()
+
 		compaignWg.Add(1)
 		go func() {
 			defer compaignWg.Done()
 
 			e.l.Debug("begin to compaign", zap.Stringer("current member", e.info))
+			if e.evictLeader.Get() {
+				// skip campaign
+				return
+			}
+
```

Comment:
How about putting it before `e.l.Debug("begin to compaign", zap.Stringer("current member", e.info))`?

Reply:
done